packages/js/product-editor/webpack.config.js

```diff
@@ -1,18 +1,72 @@
+/**
+ * External dependencies
+ */
+const CopyWebpackPlugin = require( 'copy-webpack-plugin' );
+const MiniCssExtractPlugin = require( 'mini-css-extract-plugin' );
+const path = require( 'path' );
+const RemoveEmptyScriptsPlugin = require( 'webpack-remove-empty-scripts' );
+const WebpackRTLPlugin = require( 'webpack-rtl-plugin' );
+
 /**
  * Internal dependencies
  */
 const { webpackConfig } = require( '@woocommerce/internal-style-build' );
+const {
+	blockEntryPoints,
+	getBlockMetaData,
+	getEntryPointName,
+} = require( './config/block-entry-points' );
+
+const NODE_ENV = process.env.NODE_ENV || 'development';
 
 module.exports = {
 	mode: process.env.NODE_ENV || 'development',
 	entry: {
 		'build-style': __dirname + '/src/style.scss',
+		...blockEntryPoints,
 	},
 	output: {
 		path: __dirname,
 	},
 	module: {
 		rules: webpackConfig.rules,
 	},
-	plugins: webpackConfig.plugins,
+	plugins: [
+		new RemoveEmptyScriptsPlugin(),
+		new MiniCssExtractPlugin( {
+			filename: ( data ) => {
+				return data.chunk.name.startsWith( '/build/blocks' )
+					? `[name].css`
+					: `[name]/style.css`;
+			},
+			chunkFilename: 'chunks/[id].style.css',
+		} ),
+		new WebpackRTLPlugin( {
+			test: /style\.css$/,
+			filename: '[name]/style-rtl.css',
+			minify: NODE_ENV === 'development' ? false : { safe: true },
+		} ),
+		new WebpackRTLPlugin( {
+			test: /(?<!style)\.css$/,
+			filename: '[name]-rtl.css',
+			minify: NODE_ENV === 'development' ? false : { safe: true },
+		} ),
```

Comment:
There may be an issue with one of the above items, after running a build I am seeing a `build-style-rtl.css` file generated within the `product-editor` package directory. I assume this should be placed inside the build folder?

Reply:
Ah, I see that now too.  It looks like the issue was the order of the RTL plugins in the config.  The test case for entry names was correct on both, but apparently `WebpackRTLPlugin` runs synchronously so the `build/style-rtl.css` was being created and then the plugin was creating another rtl stylesheet from that one.

Swapping the order seems to do the trick.  https://github.com/woocommerce/woocommerce/pull/37318/commits/29c59b0a1c3de7c14f897a89ca707b952f85cb4a